.github/workflows/pull-request.yaml

```diff
@@ -0,0 +1,29 @@
+name: Run Unit Tests 🚨
+
+on: [ push ]
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Setup Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 'stable'
+          check-latest: true
+
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      - name: Download Dependencies
+        run: "go mod download all"
+
+      - name: Setup etcd
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install etcd -y
+
+      - name: Unit Tests
+        run: |
```

Comment:
I think we also need to start the Etcd server right?

Reply:
The `apt-get` installation starts it. So it's already running before this step.

Reply:
Here's how they fail when I run them locally without an instance of `etcd` running
```
=== RUN   TestAdapters
{"level":"warn","ts":"2023-06-12T16:56:26.901+0530","logger":"etcd-client","caller":"v3@v3.5.0/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc00020b340/#initially=[http://127.0.0.1:2379]","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing dial tcp 127.0.0.1:2379: connect: connection refused\""}
--- FAIL: TestAdapters (5.01s)
panic: context deadline exceeded [recovered]
	panic: context deadline exceeded

goroutine 6 [running]:
testing.tRunner.func1.2({0xad9fc0, 0x10fbde0})
	/usr/local/go/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1529 +0x39f
panic({0xad9fc0, 0x10fbde0})
	/usr/local/go/src/runtime/panic.go:884 +0x213
github.com/rapyuta-robotics/etcd-adapter.(*Adapter).connect(0xc0002e4db0?)
	/home/pallab/workspace/go/src/github.com/rapyuta-robotics/etcd-adapter/adapter.go:98 +0x145
github.com/rapyuta-robotics/etcd-adapter.newAdapter({0xc0002d65a0, 0x1, 0x1}, {0xb7e9dc, 0x12})
	/home/pallab/workspace/go/src/github.com/rapyuta-robotics/etcd-adapter/adapter.go:74 +0xb6
github.com/rapyuta-robotics/etcd-adapter.NewAdapter(...)
	/home/pallab/workspace/go/src/github.com/rapyuta-robotics/etcd-adapter/adapter.go:63
github.com/rapyuta-robotics/etcd-adapter.initPolicy(0xa7cde0?, {0xb7e9dc, 0x12}, {0xc0002d65a0, 0x1, 0x1})
	/home/pallab/workspace/go/src/github.com/rapyuta-robotics/etcd-adapter/adapter_test.go:27 +0xbf
github.com/rapyuta-robotics/etcd-adapter.testSaveLoad(0xc000086760?, {0xb7e9dc, 0x12}, {0xc0002d65a0, 0x1, 0x1})
	/home/pallab/workspace/go/src/github.com/rapyuta-robotics/etcd-adapter/adapter_test.go:50 +0x4a
github.com/rapyuta-robotics/etcd-adapter.TestAdapters(0x0?)
	/home/pallab/workspace/go/src/github.com/rapyuta-robotics/etcd-adapter/adapter_test.go:112 +0x5c
testing.tRunner(0xc0002de340, 0xbc2300)
	/usr/local/go/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1629 +0x3ea
```